lib/dfid-transition/transform/attachment_collection.rb

```diff
@@ -0,0 +1,49 @@
+require 'dfid-transition/transform/attachment'
+
+module DfidTransition
+  module Transform
+    class AttachmentCollection
+      include Enumerable
+
+      attr_reader :attachments
+
+      def initialize(attachments)
+        @attachments = attachments
+      end
+
+      def each
+        @attachments.each { |attachment| yield attachment }
+      end
+
+      def downloads
+        @attachments.select(&:hosted_at_r4d?)
+      end
+
+      ##
+      # Sets the title of attachment(s) where able
+      def normalize!(document_title)
+        case
+        when count == 1
+          @attachments.first.title = document_title if @attachments.one?
+        when count == 2 && one_r4d_one_bumph?
+          @attachments = @attachments.reject(&:external_link?)
+          @attachments.first.title = document_title
+        end
+        self
+      end
```

Comment:
Theoretically there could be some. I'll need to run a basename analysis. In practice, given that the scope of uniqueness is the document, I'd think it unlikely.